lucet-runtime/lucet-runtime-internals/src/region/mmap.rs

```diff
@@ -90,8 +92,23 @@ impl RegionInternal for MmapRegion {
             lucet_bail!("heap is not page-aligned; this is a bug");
         }
 
-        let limits = &slot.limits;
-        module.validate_runtime_spec(limits)?;
+        let mut limits = slot.limits.clone();
+
+        // Affirm that any custom heap memory size supplied to this
+        // builder does not exceed the slot limits.
+        if let Ordering::Greater = heap_memory_size.cmp(&slot.limits.heap_memory_size) {
+            return Err(Error::InvalidArgument(
+                "heap memory size requested for instance is larger than slot allows",
+            ));
+        }
+
+        // If a custom heap memory size is supplied, augment the limits
+        // with the value so that it may be validated.
+        if let Ordering::Less = heap_memory_size.cmp(&slot.limits.heap_memory_size) {
+            limits.heap_memory_size = heap_memory_size;
+        }
```

Comment:
I think this would be easier to follow as a single `match heap_memory_size.cmp(...) { ... }`, and would make us be more explicit about what happens in the case that they're equal